java/src/main/native/ai_onnxruntime_OnnxTensor.c

```diff
@@ -101,32 +103,28 @@
  */
 JNIEXPORT jlong JNICALL Java_ai_onnxruntime_OnnxTensor_createString
         (JNIEnv * jniEnv, jclass jobj, jlong apiHandle, jlong allocatorHandle, jstring input) {
-    (void) jobj; // Required JNI parameter not needed by functions which don't need to access their host object.
-    const OrtApi* api = (const OrtApi*) apiHandle;
-    OrtAllocator* allocator = (OrtAllocator*) allocatorHandle;
+  (void) jobj;  // Required JNI parameter not needed by functions which don't need to access their host object.
+  const OrtApi* api = (const OrtApi*) apiHandle;
+  OrtAllocator* allocator = (OrtAllocator*) allocatorHandle;
 
-    // Extract the shape information
-    int64_t* shapeArr;
-    checkOrtStatus(jniEnv,api,api->AllocatorAlloc(allocator,sizeof(int64_t),(void**)&shapeArr));
-    shapeArr[0] = 1;
-
-    // Create the OrtValue
-    OrtValue* ortValue;
-    checkOrtStatus(jniEnv, api, api->CreateTensorAsOrtValue(allocator,shapeArr,0,ONNX_TENSOR_ELEMENT_DATA_TYPE_STRING,&ortValue));
-
-    // Release the shape
-    checkOrtStatus(jniEnv, api, api->AllocatorFree(allocator,shapeArr));
+  // Create the OrtValue
+  int64_t shapeArr = 1;
+  OrtValue* ortValue;
+  OrtErrorCode code = checkOrtStatus(jniEnv, api, api->CreateTensorAsOrtValue(allocator, &shapeArr, 0,
+                                                                              ONNX_TENSOR_ELEMENT_DATA_TYPE_STRING, &ortValue));
 
+  if (code == ORT_OK) {
     // Create the buffer for the Java string
     const char* stringBuffer = (*jniEnv)->GetStringUTFChars(jniEnv,input,NULL);
 
     // Assign the strings into the Tensor
     checkOrtStatus(jniEnv, api, api->FillStringTensor(ortValue,&stringBuffer,1));
```

Comment:
>checkOrtStatus [](http://example.com/codeflow?start=4&length=14)

The return code is not checked

Reply:
It doesn't matter in this case as we're going to clean up the Java side and return anyway, so there is no alternative action to perform on failure.

Reply:
I think there is. Imagine this.
1) OrtValue is allocated, Java strings are converted. 
2) FillStringTensor fails but ignored
3) Java strings are released
4) we return ortValue  with garbage.